docs/rfc/rfc-7.adoc

```diff
@@ -0,0 +1,201 @@
+Ethereum reveal transaction points to an existing Bitcoin P2SH transaction.
+
+==== State differences
+
+Every tBTC v2 client software instance is connected to different Ethereum and
+Bitcoin clients. Ethereum clients in the network will have a slightly different
+synchronization state at any given point in time. Bitcoin clients in the network
+will also have a slightly different synchronization state. Since the light relay
+is not submitting all Bitcoin headers to Ethereum, there is no way to reference
+from Ethereum to a fixed point in time on Bitcoin. We must accept all tBTC v2
+client software will have a slightly different view of the world at any point in
+time.
+
+==== Adhere to network conditions
+
+The Bitcoin sweep transaction must have a fee that allows it to be mined in a
+reasonable time. If the Bitcoin network conditions change, there must be a
+mechanism to increase the fee and replace transactions. The wallet must finish
+the current operation before attempting another one.
+
+==== Retries
+
+The tECDSA protocol used for signing must be able to retry in case some signing
+group members are offline or corrupted. The tBTC v2 client software should join
+the pending signing after the client restart.
+
+=== Implementation
+
+==== Coordinator
+
+We introduce a rotating role of Coordinator responsible for proposing a sweep
+Bitcoin transaction. The role of Coordinator rotates every 6 hours across all
+operators in the group. For example, if the group consist of the following
+members: `[0xA, 0xB, 0xC, 0xC, 0xB, 0xD`] the rotation rounds are: `0xA, 0xB,
+0xC, 0xD`. The rotation schedule begins at UNIX epoch (00:00:00 UTC on 1 January
+1970).
+
+==== Sweeping schedule
+
+Sweep can only be proposed by the current Coordinator. The Coordinator can only
+propose a sweep between:
+
+- 00:00 UTC - 0:30 UTC
+- 08:00 UTC - 08:30 UTC
+- 16:00 UTC - 16:30 UTC
+
+If the current Coordinator does not propose a sweep within the given time
+window, the attempt is skipped.
+
+
+==== Sweep transaction
+
+The Coordinator chooses sufficiently confirmed deposits for the sweep. To avoid
+synchronization problems with deposits on the brink of being sufficiently
+confirmed, the Coordinator may choose deposits with more than 6/20 confirmations.
+To avoid problems with a maximum size of a transaction, the Coordinator should
+not pick more than 500 deposits. The Coordinator chooses Bitcoin transaction fee
+reflecting the current situation in the Bitcoin network.
+
+The Coordinator must not propose a sweep if there is another transaction from
+the wallet in the Bitcoin mempool.
+
+The Coordinator proposes a sweep over the wallet's broadcast channel.
+
+All signing group members must confirm that all deposits in the proposed sweep
+are sufficiently confirmed, that the maximum number of deposits is not exceeded,
+and that the proposed fee reflects the current situation in the Bitcoin network
+plus/minus some margin.
```

Comment:
All signing group members should also make sure that all UTXOs produced by the selected P2(W)SH deposit transactions can be unlocked using the wallet's public key.

Reply:
See c0d2950d0b96fab1fdaf0e51231ff660e3985976, please.